### ISSUE TITLE
Pre-training using GPUs

### ISSUE THREAD

**Issue by key2miao** [s3prl/s3prl#72]
Hello, 

Thank you for your efforts on this cool work. I would like to know what and how many GPUs could support for pre-training model? (like GPU memory and which kind of GPU at least).  Many thanks.

**Comment by andi611** [comment]
Hi,

We have this cool feature called `gradient_accumulate_steps`, where you can accumulate gradients to form a larger batch size.
Hence the GPU memory is not so important, what may concern you is how long the time it will take.
The minimal requirement is 1-GPU with at least 6GB of memory, but it will take quite some time to train.

For your reference:
With the default TERA model setting, a 2080Ti with 11GB memory can hold up to a batch size of 8.
Currently, I have a 3-layer model training on two 2080Ti, with a batch size of 8, accumulated 4 times (this is equivalent to a batch size of 32), it takes approximately 200hrs to complete 1M training steps on 960hr of audio data.

**Comment by key2miao** [comment]
> Hi,
> 
> We have this cool feature called `gradient_accumulate_steps`, where you can accumulate gradients to form a larger batch size.
> Hence the GPU memory is not so important, what may concern you is how long the time it will take.
> The minimal requirement is 1-GPU with at least 6GB of memory, but it will take quite some time to train.
> 
> For your reference:
> With the default TERA model setting, a 2080Ti with 11GB memory can hold up to a batch size of 8.
> Currently, I have a 3-layer model training on two 2080Ti, with a batch size of 8, accumulated 4 times (this is equivalent to a batch size of 32), it takes approximately 200hrs to complete 1M training steps on 960hr of audio data.

**Comment by key2miao** [comment]
Hi, if I would like to train the upstream model from scratch, which commands should be used? Many thanks.
